rio/src/org/usfirst/frc/team5806/robot/CameraClient.java

```diff
@@ -0,0 +1,78 @@
+import java.net.*;
+import java.io.*;
+
+public class CameraClient {
+	private String serverName;
+	private INetAddress staticIP;
+	private int serverPort;
+	private Socket client;
+	private BufferedWriter out;
+	private BufferedReader in;
+	
+	public CameraClient(String serverName, int serverPort) {
+		this.serverName = serverName;
+		this.serverPort = serverPort;
+	}
+	public CameraClient(String staticIP, int serverPort) {
+		//Alternative constructor for referencing cam by a static ip rather
+		//than a saved name on the LAN.
+		this.staticIP = INetAddress.getByAddress(staticIP);
+		this.serverPort = serverPort;
+	}
+	public CameraClient() {}
+	public void setServerName(String name) {
+		this.serverName = name;
+	}
+	public void setStaticIP(String staticIP) {
+		this.staticIP = INetAddress.getByAddress(staticIP);
+	}
+	public void setServerPort(int port) {
+		this.serverPort = port;
+	}
+	public String getServerName() {
+		return serverName;
+	}
+	public String getStaticIPName() {
+		return staticIP.toString();
+	}
+	public int getServerPort() {
+		return serverPort;
+	}
+	public void initSocket() {
+		try {
+			if (INetAddress == null) {
+				this.client = new Socket(serverName, serverPort);
+			} else {
+				this.client = new Socket(staticIP, serverPort);
+			}
+			this.out = new BufferedWriter(new OutputStreamWriter(this.client.getOutputStream()));
+			this.in = new BufferedReader(new InputStreamReader(this.client.getInputStream()));
+		} catch (IOException e) {
+			System.out.println("Error Initializing Socket or Input/Output Streams");
+		}
+		System.out.println("Client socket successfully initialized.");
+	}
+	private String readLine() {
+		return in.readLine();
+	}
+	private int[] readLineToIntArray() {
+		//The image is being passed as stream of unsigned chars
```

Comment:
You shouldnt do the whole string thing. Pretty sure it'll be slow. They are sending you an array of unsigned chars (aka. bytes in java). So, you should read an array of unsigned chars.